pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -367,20 +367,22 @@ func (h *NetworkUtilsHandler) StartDHCP(nic *VIF, serverAddr net.IP, bridgeInter
 		}
 	}()
 
-	go func() {
-		if err = DHCPv6Server(
-			nic.IPv6.IP,
-			bridgeInterfaceName,
-			nameservers,
-			nic.Routes,
-			searchDomains,
-			nic.Mtu,
-			dhcpOptions,
-		); err != nil {
-			log.Log.Errorf("failed to run DHCP: %v", err)
-			panic(err)
-		}
-	}()
+	if nic.IPv6.IPNet != nil {
```

Comment:
- Please explain "why" in the commit message.
- This condition is hard to relate to the masquerade binding. How can we make it clearer? And how can we be sure this will still be correct when code changes in the future? The indirect check is worrying.

Reply:
* What do you mean by explain "why", th commit message says, to avoid running the dhcpv6 server for masquerade. What info is missing?
* If `StartDHCP` is called with ipv6 information in the vif, it means a dhcpv6 server should run. Currently only masquerade has ipv6 information in the vif. But if any other binding with add the ipv6 to their vif, it makes sense the dhcpv6 server will run. Do you think the code should be explicit and allow running of the dhcpv6 server only for masquerade?

Reply:
Done. Updated the commit message.